### PR TITLE
[WIP] ENT-134: Display consent to share data for Coupon use

### PR DESCRIPTION
**Pull request by mtyaka**
**Description:** 

This PR is not complete yet. More info to follow.

**JIRA:** [ENT-134](https://openedx.atlassian.net/browse/ENT-134)

**Testing instructions:**

TBD

**Reviewers:**
- [ ] OpenCraft: @haikuginger 
- [ ] edX: TBD 

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Commits are (reasonably) squashed
- [ ] Translations are updated
- [ ] PR author is listed in AUTHORS

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPi after tag-triggered build is 
      finished.
- [ ] Delete working branch (if not needed anymore)

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @mtyaka! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

<a   href="https://openedx-webhooks.herokuapp.com/github/process_pr?repo=edx%2Fedx-enterprise&number=36"   rel="nofollow" >Create an OSPR issue for this pull request</a>.

**Comment by mattdrayer** [comment]
@zubair-arbi @saleem-latif @asadiqbal08 FYI on this PR

File: enterprise/api/v1/views.py [hunk 1]
```diff
@@ -27,6 +27,14 @@ class EnterpriseReadOnlyModelViewSet(viewsets.ReadOnlyModelViewSet):
     throttle_classes = (ServiceUserThrottle,)
 
 
+class EnterpriseCreateListRetrieveUpdateModelViewSet(
```
Comment (saleem-latif) [inline enterprise/api/v1/views.py]: From View name it is not clear which models this view set belongs to, can you update the name to something else ?

Comment (saleem-latif) [inline enterprise/api/v1/views.py]: Oh, I get it, this is the base class for enterprise View sets.

**Comment by asadiqbal08** [comment]
@haikuginger Is there anything required in that PR to complete ? I am working on ENT-115 that is dependent on that PR.

**Comment by haikuginger** [comment]
@asadiqbal08, we're still working on ENT-134, which takes advantage of this; because there _may_ be changes required as part of that story, we'd like to wait until we've got that completed.

File: enterprise/api/v1/views.py (hunk 1 above)
Comment (mattdrayer) [inline enterprise/api/v1/views.py]: Would it be simpler to name this `EnterpriseReadWriteModelViewSet`?

**Comment by bradenmacdonald** [comment]
@mtyaka @haikuginger Does this PR restrict API access to staff users only? From a quick look I see only the "IsAuthenticated" check, which does not seem sufficient.

@haikuginger @bdero Modulo the above, are we ready to merge this now? I'd expect the requirements around ENT-134 have been finalized...

**Comment by bradenmacdonald** [comment]
Also, question on this PR: Does this allow creation of EnterpriseCourseEnrollments even if the "real" course enrollment does not exist, or the course does not exist?

**Comment by haikuginger** [comment]
@bradenmacdonald, good catch. Yeah, restricting to staff users would be a good bet. And yeah, if @bdero is set on his needs, then let's go ahead with getting this ready to merge @mtyaka.

**Comment by mattdrayer** [comment]
@bradenmacdonald @mtyaka @bdero @haikuginger -- I just have one question about a View name otherwise this PR looks good to me -- 👍

**Comment by mtyaka** [comment]
Closing this in favor of #40.

**Pull request closed without merging**